catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserver.java

```diff
@@ -0,0 +1,310 @@
+            }
+        }
+        final DavEntry[] children = entry.getChildren();
+        for (final DavEntry aChildren : children) {
+            doCreate(aChildren);
+        }
+    }
+
+    /**
+     * Fire directory/file change events to the registered listeners.
+     *
+     * @param entry The previous file system entry
+     * @param file  The current file
+     */
+    private void doMatch(final DavEntry entry, final DavResource file) {
+        if (entry.refresh(file)) {
+            for (final EntryAlterationListener listener : listeners) {
+                if (entry.isDirectory()) {
+                    listener.onDirectoryChange(entry);
+                } else {
+                    listener.onFileChange(entry);
+                }
+            }
+        }
+    }
+
+    /**
+     * Fire directory/file delete events to the registered listeners.
+     *
+     * @param entry The file entry
+     */
+    private void doDelete(final DavEntry entry) {
+        for (final EntryAlterationListener listener : listeners) {
+            if (entry.isDirectory()) {
+                listener.onDirectoryDelete(entry);
+            } else {
+                listener.onFileDelete(entry);
+            }
+        }
+    }
+
+    /**
+     * List the contents of a directory
+     *
+     * @param file The file to list the contents of
+     * @return the directory contents or a zero length array if
+     * the empty or the file is not a directory
+     */
+    private DavResource[] listFiles(final String file) {
+        DavResource[] children = null;
+        try {
+            List<DavResource> list = sardine.list(file);
+            // the returned list includes the parent
+            if (list.size() > 1 && list.get(0)
+                    .isDirectory()) {
+                children = list.subList(1, list.size())
+                        .toArray(new DavResource[list.size() - 1]);
+                List<DavResource> resourceList = Arrays.asList(children);
+                // lexicographical sorting
+                resourceList.sort(Comparator.comparing(DavResource::getName));
+                children = resourceList.toArray(new DavResource[resourceList.size()]);
+            }
+        } catch (IOException e) {
+            // if it doesn't exist it can't have children
+            children = EMPTY_RESOURCES;
+        }
+        if (children == null) {
+            children = EMPTY_RESOURCES;
+        }
+        return children;
+    }
+
+    /**
+     * Provide a String representation of this observer.
+     *
+     * @return a String representation of this observer
+     */
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "[file='" + getDirectory() + '\'' + ", listeners="
+                + listeners.size() + "]";
+    }
+
+    private void readObject(java.io.ObjectInputStream in)
+            throws IOException, ClassNotFoundException {
+        in.defaultReadObject();
```

Comment:
I don't think that's necessary here. The serialization is stored in `DDF_HOME/data` so an attacker would have to access to the filesystem, at which point there's much worse things they could be doing. We've always treated the installation directory as secure.